backend/infrahub/cli/db.py

```diff
@@ -185,98 +192,109 @@ async def update_core_schema(  # pylint: disable=too-many-statements
-        schema = SchemaRoot(**internal_schema)
-        registry.schema.register_schema(schema=schema)
-
-        # ----------------------------------------------------------
-        # Load Current Schema from the database
-        # ----------------------------------------------------------
-        schema_default_branch = await registry.schema.load_schema_from_db(db=db, branch=default_branch)
-        registry.schema.set_schema_branch(name=default_branch.name, schema=schema_default_branch)
-        branch_schema = registry.schema.get_schema_branch(name=registry.default_branch)
-
-        candidate_schema = branch_schema.duplicate()
-        candidate_schema.load_schema(schema=SchemaRoot(**internal_schema))
-        candidate_schema.load_schema(schema=SchemaRoot(**core_models))
-        candidate_schema.load_schema(schema=SchemaRoot(**deprecated_models))
-        candidate_schema.process()
-
-        result = branch_schema.validate_update(other=candidate_schema, enforce_update_support=False)
-        if result.errors:
-            rprint(f"{error_badge} | Unable to update the schema, due to failed validations")
-            for error in result.errors:
-                rprint(error.to_string())
-            raise typer.Exit(1)
-
-        if not result.diff.all:
-            rprint("Core Schema Up to date, nothing to update")
-            raise typer.Exit(0)
-
-        rprint("Core Schema has diff, will need to be updated")
-        if debug:
-            result.diff.print()
-
-        # ----------------------------------------------------------
-        # Validate if the new schema is valid with the content of the database
-        # ----------------------------------------------------------
-        error_messages, _ = await schema_validators_checker(
-            branch=default_branch, schema=candidate_schema, constraints=result.constraints, service=service
-        )
-        if error_messages:
-            rprint(f"{error_badge} | Unable to update the schema, due to failed validations")
-            for message in error_messages:
-                rprint(message)
-            raise typer.Exit(1)
-
-        # ----------------------------------------------------------
-        # Update the schema
-        # ----------------------------------------------------------
-        origin_schema = branch_schema.duplicate()
-
-        # Update the internal schema
-        schema_default_branch.load_schema(schema=SchemaRoot(**internal_schema))
-        schema_default_branch.process()
-        registry.schema.set_schema_branch(name=default_branch.name, schema=schema_default_branch)
-
-        async with db.start_transaction() as dbt:
-            await registry.schema.update_schema_branch(
-                schema=candidate_schema,
-                db=dbt,
-                branch=default_branch.name,
-                diff=result.diff,
-                limit=result.diff.all,
-                update_db=True,
+        with prefect_test_harness():
```

Comment:
Yes, by default flow and task will report their status to a prefect server. 
`prefect_test_harness` start a server locally and all flows and tasks executed within the context manager will automatically use this server.
it's helpful to reuse the same task in a pure CLI mode without external dependencies.

having said that, I was having second thoughts about this one, it feels like it could be useful to have more visibility about the execution of this task in prefect for troubleshooting etc ...